data/block/block.go

```diff
@@ -581,3 +582,39 @@ func (h *Header) GetAdditionalData() headerVersionData.HeaderAdditionalData {
 	// no extra data for the initial version of shard block header
 	return nil
 }
+
+// CheckFieldsForNil checks a predefined set of fields for nil values
+func (h *Header) CheckFieldsForNil() error {
+	if h == nil {
+		return data.ErrNilPointerReceiver
+	}
+	if h.PrevHash == nil {
```

Comment:
PrevHash, PrevRandSeed and others fields checked here of the genesis block are not nil?

Reply:
as seen here, no, they are populated: https://github.com/multiversx/mx-chain-go/blob/10c4bef8b57431eee74fa0d34af29c86a05f58c6/genesis/process/shardGenesisBlockCreator.go#L207
https://github.com/multiversx/mx-chain-go/blob/10c4bef8b57431eee74fa0d34af29c86a05f58c6/genesis/process/metaGenesisBlockCreator.go#L97

Reply:
ok